src/binary-reader-objdump.cc

```diff
@@ -1840,16 +1842,50 @@ Result BinaryReaderObjdump::PrintSymbolFlags(uint32_t flags) {
       vis_name = "default";
       break;
   }
-  if (flags & WABT_SYMBOL_FLAG_UNDEFINED)
+  flags &= ~WABT_SYMBOL_MASK_VISIBILITY;
+
+  PrintDetails(" [");
+  if (flags & WABT_SYMBOL_FLAG_UNDEFINED) {
     PrintDetails(" undefined");
-  if (flags & WABT_SYMBOL_FLAG_EXPORTED)
+    flags &= ~WABT_SYMBOL_FLAG_UNDEFINED;
+  }
+  if (flags & WABT_SYMBOL_FLAG_EXPORTED) {
     PrintDetails(" exported");
-  if (flags & WABT_SYMBOL_FLAG_EXPLICIT_NAME)
+    flags &= ~WABT_SYMBOL_FLAG_EXPORTED;
+  }
+  if (flags & WABT_SYMBOL_FLAG_EXPLICIT_NAME) {
     PrintDetails(" explicit_name");
-  if (flags & WABT_SYMBOL_FLAG_NO_STRIP)
+    flags &= ~WABT_SYMBOL_FLAG_EXPLICIT_NAME;
+  }
+  if (flags & WABT_SYMBOL_FLAG_NO_STRIP) {
     PrintDetails(" no_strip");
+    flags &= ~WABT_SYMBOL_FLAG_NO_STRIP;
+  }
+  if (flags != 0) {
+    PrintDetails(" %d", flags);
```

Comment:
so the idea is that this just prints leftover/unknown flags? If so, why not make it explicitly say it's unknown, as you do with segment flags?

Reply:
Yeah.. I agree.  Will do.